pipelines/components/transform_android.py

```diff
@@ -59,21 +66,28 @@ def map_field(field, version):
             else:
                 return f"ds.{version}Measurement.{field['name']} "
 
-        current_fields = map(lambda field: map_field(field, 'current'), android_fields)
+        project = events_table.split(".")[1]
+        table = events_table.split(".")[2]
+        table_ref = bigquery.DatasetReference(project_name, project).table(table)
+        struct_schema = bigquery_client.get_table(table_ref).schema
+        struct_schema_fields = [field.name for field in struct_schema[1].fields]
+        sorted_struct_schema_fields = sorted(android_fields, key=lambda e: struct_schema_fields.index(e["name"]))
+
+        current_fields = map(lambda field: map_field(field, 'current'), sorted_struct_schema_fields)
         current_fields_str = ", ".join(current_fields)
 
-        previous_fields = map(lambda field: map_field(field, 'previous'), android_fields)
+        previous_fields = map(lambda field: map_field(field, 'previous'), sorted_struct_schema_fields)
         previous_fields_str = ", ".join(previous_fields)
 
         current_hash_input_fields = [f"CAST(ds.currentMeasurement.`{field['from_name']}` AS STRING)" for field in
-                                     android_fields if field['from_name'] != 'Stats_timestamp']
+                                     android_fields if field["hash"]]
```

Comment:
Just a double check: I think I see what you did here, by introducing the "hash" field. This allows you to compare hashes, despite the version of the library, right?
So that's nice.
But does this make sense in the long run? Because we do leave some valid fields out of the hash. Or doesn't that really matter, since changes of collision are very small already?

Reply:
Originally we excluded only `Stats_timestamp` when creating the hash. If we update the schema without the above functionality the hashes for previous fields would be different than when those previous events were inserted. This is because new fields with a value of null would be included in the hash and thus an incomparable values for the hash would be the result. 
The conclusion would be a transitioning phase where any comparison between the old and new schema would lead to duplicate entries in the aggregation table. This of course isn't the intended behavior.
A possible fix would be to retroactively calculate the hashes for all previous events and update accordingly. I feel like this is quite a bit of hassle and it's better to flag the fields that need to be hashed. The valid fields aren't very important because, like you said, the chance of collision is very slim. This chance seems so slim that the risk of incorrect deletion isn't significant.